server/src/main/java/org/opensearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -283,6 +287,10 @@ public void innerOnResponse(Result result) {
                             } finally {
                                 executeNext(pendingExecutions, thread);
                             }
+                            ThreadMXBean threadMXBean = (ThreadMXBean) ManagementFactory.getThreadMXBean();
+                            long bytes = threadMXBean.getThreadAllocatedBytes(Thread.currentThread().getId());
+
+                            TaskResourceTracker.getInstance().transfer(task.getId(), result, bytes);
```

Comment:
It would be too hard to maintain the code base with this construct. Lets simplify